### ISSUE TITLE
semicolon after return causes internal compiler error

### ISSUE THREAD

**Issue by hishamhm** [teal-language/tl#495]
As noted here by @Ghost4Man: https://github.com/teal-language/tl/issues/391#issuecomment-974422577

Test case:

```lua
-- ice.tl
local function get_foo(): string
   if true then
      return "foo";
      print("unreachable")
   end
end
```

fails with `tl run ice.tl`

```
Internal Compiler Error: Teal generator produced invalid Lua. Please report a bug at https://github.com/teal-language/tl

ice.tl:4: 'end' expected (to close 'if' at line 2) near 'print'
```

Also, `tl check ice.tl` reports no errors. If you remove the semicolon, it produces a `syntax error`, as expected (and seen in #391).

The solution to this problem is to make the case above _produce_ a syntax error, so that it never gets to the code generation phase generating invalid Lua that contains code after a `return`.

**Comment by HamzaMateen** [comment]
Would love to work on this but seems like it's pretty old, can I undertake this? thank you!

**Comment by hishamhm** [comment]
@HamzaMateen yes, you can! This is still an active bug. It's a good way to start getting acquainted with the Teal parser.

**Comment by HamzaMateen** [comment]
great, I will start working on it, thank you!
